src/sentry/profiles/task.py

```diff
@@ -273,37 +306,67 @@ def _normalize(profile: Profile, organization: Organization) -> None:
             profile["device_classification"] = classification
 
 
-def _prepare_frames_from_profile(profile: Profile) -> Tuple[List[Any], List[Any], set[int]]:
+def _prepare_frames_from_profile(
+    profile: Profile, platform: Optional[str] = None
```

Comment:
The platform here should be required from what I can tell given that the caller always passes a platform to this function

Reply:
yep, I've originally left it that way for not having to refactor the tests. But looking into the tests it's just a couple of them and the edit is trivial so I've made it mandatory as you suggested. 👍🏻